### PR TITLE
macpackage module: expand user before mounting

### PR DESCRIPTION
**Pull request by dschaller**
### What does this PR do?
Changes the `macpackage` module to expand the user substitution in a dmg path before attempting to mount it.

### What issues does this PR fix or reference?
Fixes: currently, if you try to mount any dmg that has a shorthand for the users's home (e.g. `~`) `hdiutil attach` will fail.

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [x] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
No

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Review by Ch3LL — CHANGES_REQUESTED** [review]
This will require a changelog and test coverage

**Comment by dschaller** [comment]
@Ch3LL thanks for the first pass. Updated and should be ready for another look when you have a minute

File: changelog/63905.fixed.md [hunk 1]
```diff
@@ -0,0 +1 @@
+added proper expanding of user directory when mounting a macpackage
```
Comment (Ch3LL) [inline changelog/63905.fixed.md:1]: 63905 is your other PR, can you update this to 63894

**Review by Ch3LL — CHANGES_REQUESTED** [review]

**Comment by dwoz** [comment]
Closing this due to inactivity. Anyone should feel free to re-open it if they want to see it through to the end in one release cycle.

@dschaller Feel free to re-open this if you want to get it into a release.

**Pull request closed without merging**